libpromises/evalfunction.c

```diff
@@ -8136,6 +8142,45 @@ void ModuleProtocol(EvalContext *ctx, char *command, const char *line, int print
         }
         break;
 
+    case '&':
+        // TODO: the variable name is limited to 256 to accommodate the
+        // context name once it's in the vartable.  Maybe this can be relaxed.
+        // &policy_varname=/path/to/file can be json/env/yaml/csv type, default: json
+        sscanf(line + 1, "%256[^=]=%4095[^\n]", name, content);
+
+        if (CheckID(name))
+        {
+            if (FileCanOpen(content, "r"))
+            {
+                const int size_max = IntFromString("inf");
+                const char *requested_mode = FileType(content);
+
+                Log(LOG_LEVEL_DEBUG, "Module protocol parsing %s file '%s'", requested_mode, content);
+
+                JsonElement *json = JsonReadDataFile("module file protocol", content, requested_mode, size_max);
+                if (json != NULL)
+                {
+                    Buffer *tagbuf = StringSetToBuffer(tags, ',');
+                    VarRef *ref = VarRefParseFromScope(name, context);
+
+                    EvalContextVariablePut(ctx, ref, json, CF_DATA_TYPE_CONTAINER, BufferData(tagbuf));
+                    VarRefDestroy(ref);
+                    BufferDestroy(tagbuf);
+                    JsonDestroy(json);
+                }
+                else
+                {
+                    // reading / parsing failed, error message printed by JsonReadDataFile,
+                    // variable will not be created
+                }
+            }
+            else
+            {
+                Log(LOG_LEVEL_VERBOSE, "could not load module file '%s'", content);
```

Comment:
I think this should be an error.

Reply:
Can change it to `ERROR`. I have copied the code from `LoadAugmentsFiles`. There it is a `VERBOSE` message if the json file does not exists.

Reply:
Yes, but in loading augments I think it is because of cases where a variable is used in file name and similar where it is kind of expected that the file doesn't exits. Here it shouldn't happen, right?